pkg/retrieval/retrieval.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:generate sh -c "protoc -I . -I \"$(go list -f '{{ .Dir }}' -m github.com/gogo/protobuf)/protobuf\" --gogofaster_out=. retrieval.proto"
+
+package retrieval
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/storage"
+)
+
+const (
+	protocolName  = "retrieval"
+	streamName    = "retrieval"
+	streamVersion = "1.0.0"
+)
+
+type Service struct {
+	streamer      p2p.Streamer
+	peerSuggester p2p.PeerSuggester
+	storer        storage.Storer
+	logger        Logger
+}
+
+type Options struct {
+	Streamer      p2p.Streamer
+	PeerSuggester p2p.PeerSuggester
+	Storer        storage.Storer
+	Logger        Logger
```

Comment:
`logging.Logger` has recently been added and should be used here instead the local interface. Just a suggestion.